### PR TITLE
User tokens

### PR DESCRIPTION
**Pull request by ian-fox**
* `GET /tokens` will return a list of all tokens belonging to the current user

### CODE REVIEW

File: plugins/tokens/list.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+            return userFactory.get({ username })
+            .then((user) => {
+                if (!user) {
+                    throw boom.notFound(`User ${username} does not exist`);
+                }
+
+                return user.tokens;
+            })
+            .then(tokens => reply(tokens.map((t) => {
```
Comment (FenrirUnbound) [inline plugins/tokens/list.js]: Preference to using meaningful variable names

File: test/plugins/tokens.test.js [hunk 2]
```diff
@@ -52,7 +52,8 @@ describe('token plugin test', () => {
     beforeEach((done) => {
         tokenFactoryMock = {
             create: sinon.stub(),
-            get: sinon.stub()
+            get: sinon.stub(),
+            list: sinon.stub()
```
Comment (FenrirUnbound) [inline test/plugins/tokens.test.js]: This doesn't seem to be used

Comment (FenrirUnbound) [inline test/plugins/tokens.test.js]: In fact, I don't see where having the `tokenFactoryMock` is significant

File: plugins/tokens/list.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                delete output.userId;
+
+                return output;
+            })))
+            .catch(err => reply(boom.wrap(err)));
+        },
+        response: {
+            schema: getSchema
+        },
+        validate: {
```
Comment (FenrirUnbound) [inline plugins/tokens/list.js]: I don't think route handles pagination just yet

File: test/plugins/tokens.test.js (hunk 2 above)
Comment (ian-fox) [inline test/plugins/tokens.test.js]: Ah, right, because `userFactory` is mocked

**Pull request merged**